src/Language/Fixpoint/Solver/Solve.hs

```diff
@@ -110,12 +125,26 @@ solve_ cfg fi s0 ks wkl = do
     res0     <- {- SCC "sol-result" #-} result bindingsInSmt cfg wkl s3
     return (s3, res0)
   res <- case resStatus res0 of
-    Unsafe _ bads | not (noLazyPLE cfg) && rewriteAxioms cfg -> do
+    Unsafe _ bads | not (noLazyPLE cfg) && rewriteAxioms cfg && useInterpreter cfg -> do
```

Comment:
Is it necessary to choose between the two PLE implementations? I would have expected a first pass with the interpreter, and then a second pass on the remaining unsolved constraints with the smt-based PLE.